great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -724,7 +724,8 @@ def delete_datasource(
             raise ValueError(f"Datasource {datasource_name} not found")
 
         if save_changes:
-            self._datasource_store.delete_by_name(datasource_name)  # type: ignore[attr-defined]
+            datasource_config = datasourceConfigSchema.load(datasource.config)
+            self._datasource_store.delete(datasource_config)  # type: ignore[attr-defined]
```

Comment:
We need a way to pass down the `id` to the store - using the config is the cleanest way (as opposed to just name)